main/src/ca/uwaterloo/flix/language/phase/OccurrenceAnalyzer.scala

```diff
@@ -254,6 +254,10 @@ object OccurrenceAnalyzer {
     case Expression.Region(tpe, loc) =>
       (OccurrenceAst.Expression.Region(tpe, loc), OccurInfo.One)
 
+    case Expression.Scope(sym, exp, tpe, purity, loc) =>
+      val (e, o) = visitExp(sym0, exp)
+      (OccurrenceAst.Expression.Scope(sym, e, tpe, purity, loc), o.copy(defs = o.defs + (sym0 -> DontInline)).increaseSizeByOne())
```

Comment:
I would prefer a strong comment not to inline regions. But I guess that can be added in the next PR.